lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -5,45 +5,46 @@ defmodule ExUnit.Runner do
 
   @rand_algorithm :exs1024
 
-  def run(opts, load_us) do
-    {:ok, manager} = EM.start_link()
-    {:ok, stats} = EM.add_handler(manager, ExUnit.RunnerStats, opts)
-    {opts, config} = configure(manager, opts)
+  @type manager :: ExUnit.EventManager.manager()
+
+  def run(load_time, opts) when (is_integer(load_time) or is_nil(load_time)) and is_list(opts) do
+    opts = normalize_opts(opts)
 
+    {:ok, manager} = EM.start_link()
+    {:ok, stats_pid} = EM.add_handler(manager, ExUnit.RunnerStats, opts)
+    config = configure(opts, manager, self(), stats_pid)
     :erlang.system_flag(:backtrace_depth, Keyword.fetch!(opts, :stacktrace_depth))
 
-    {run_us, _} =
+    {run_time, _} =
       :timer.tc(fn ->
         EM.suite_started(config.manager, opts)
-        loop(config, 0)
+        loop(config)
       end)
 
-    EM.suite_finished(config.manager, run_us, load_us)
-    result = ExUnit.RunnerStats.stats(stats)
+    EM.suite_finished(config.manager, run_time, load_time)
+    stats = ExUnit.RunnerStats.stats(stats_pid)
     EM.stop(config.manager)
     after_suite_callbacks = Application.fetch_env!(:ex_unit, :after_suite)
-    Enum.each(after_suite_callbacks, fn callback -> callback.(result) end)
-    result
+    Enum.each(after_suite_callbacks, fn callback -> callback.(stats) end)
+    stats
   end
 
-  defp configure(manager, opts) do
-    opts = normalize_opts(opts)
+  defp configure(opts, manager, runner_pid, stats_pid) do
     Enum.each(opts[:formatters], &EM.add_handler(manager, &1, opts))
 
-    config = %{
+    %{
       capture_log: opts[:capture_log],
       exclude: opts[:exclude],
       include: opts[:include],
       manager: manager,
       max_cases: opts[:max_cases],
       only_test_ids: opts[:only_test_ids],
+      runner_pid: runner_pid,
       seed: opts[:seed],
-      modules: :async,
```

Comment:
so now, `modules` is passed around by the loop function, as there is no need to keep it in config.